modules/stereo/src/Descriptor.cpp

```diff
@@ -0,0 +1,448 @@
+//Copyright (C) 2009-2011, Willow Garage Inc., all rights reserved.
+//Copyright (C) 2009-2015, NVIDIA Corporation, all rights reserved.
+//Copyright (C) 2010-2013, Advanced Micro Devices, Inc., all rights reserved.
+//Copyright (C) 2015, OpenCV Foundation, all rights reserved.
+//Copyright (C) 2015, Itseez Inc., all rights reserved.
+//Third party copyrights are property of their respective owners.
+//
+//Redistribution and use in source and binary forms, with or without modification,
+//are permitted provided that the following conditions are met:
+//
+//  * Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//
+//  * Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+//  * Neither the names of the copyright holders nor the names of the contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+//This software is provided by the copyright holders and contributors "as is" and
+//any express or implied warranties, including, but not limited to, the implied
+//warranties of merchantability and fitness for a particular purpose are disclaimed.
+//In no event shall copyright holders or contributors be liable for any direct,
+//indirect, incidental, special, exemplary, or consequential damages
+//(including, but not limited to, procurement of substitute goods or services;
+//loss of use, data, or profits; or business interruption) however caused
+//and on any theory of liability, whether in contract, strict liability,
+//or tort (including negligence or otherwise) arising in any way out of
+//the use of this software, even if advised of the possibility of such damage.
+
+/*****************************************************************************************************************\
+*   The interface contains the main descriptors that will be implemented in the descriptor class			      *
+*																												  *
+\******************************************************************************************************************/
+
+
+#include "Descriptor.hpp"
+
+using namespace cv;
+using namespace stereo;
+
+Descriptor::Descriptor(int k)
+{
+	this->kernelSize = k;
+}
+//SSE function for computing the census transform on 1 image
+void Descriptor::applyCensusOnImage_SSE(uint8_t * image, int height, int width, uint32_t * &dist)
+{
+	//TO DO
+	UNUSED( image );
+	UNUSED( height );
+	UNUSED( width );
+	UNUSED( dist );
+	
+}
+//Implementation for computing the Census transform on the given image
+void Descriptor::applyCensusOnImage(uint8_t *image, int height, int width, uint32_t * &dist)
+{
+	int n2 = (this->kernelSize - 1) / 2;
+
+//#pragma omp parallel for
```

Comment:
It's better to use OpenCV parallel for from the very beginning, so we don't forget about it later.